public/app/features/dashboard/api/dashboard_api.ts

```diff
@@ -0,0 +1,98 @@
+import { config, getBackendSrv } from '@grafana/runtime';
+import { ScopedResourceClient } from 'app/features/apiserver/client';
+import { ResourceClient } from 'app/features/apiserver/types';
+import { SaveDashboardCommand } from 'app/features/dashboard/components/SaveDashboard/types';
+import { dashboardWatcher } from 'app/features/live/dashboard/dashboardWatcher';
+import { DeleteDashboardResponse } from 'app/features/manage-dashboards/types';
+import { DashboardDTO, DashboardDataDTO } from 'app/types';
+
+export interface DashboardAPI {
+  dashboardExists(uid: string): Promise<boolean>;
+  getDashboardDTO(uid: string): Promise<DashboardDTO>;
+  deleteDashboard(uid: string, showSuccessAlert: boolean): Promise<DeleteDashboardResponse>;
+  saveDashboard(options: SaveDashboardCommand): Promise<unknown>;
+}
+
+// Implemented using /api/dashboards/*
+class LegacyDashboardAPI implements DashboardAPI {
+  constructor() {}
+
+  saveDashboard(options: SaveDashboardCommand): Promise<unknown> {
+    dashboardWatcher.ignoreNextSave();
+
+    return getBackendSrv().post('/api/dashboards/db/', {
+      dashboard: options.dashboard,
+      message: options.message ?? '',
+      overwrite: options.overwrite ?? false,
+      folderUid: options.folderUid,
+    });
+  }
+
+  dashboardExists(uid: string): Promise<boolean> {
+    throw new Error('Method not implemented.');
+  }
+
+  deleteDashboard(uid: string, showSuccessAlert: boolean): Promise<DeleteDashboardResponse> {
+    return getBackendSrv().delete<DeleteDashboardResponse>(
+      `/api/dashboards/uid/${uid}`, { showSuccessAlert });
+  }
+
+  getDashboardDTO(uid: string): Promise<DashboardDTO> {
+    return getBackendSrv().get<DashboardDTO>(`/api/dashboards/uid/${uid}`);
+  }
+}
+
+// Implemented using /apis/dashboards.grafana.app/*
+class K8sDashboardAPI implements DashboardAPI {
+  private client: ResourceClient<DashboardDataDTO>;
+  constructor(private legacy: DashboardAPI) {
+    this.client = new ScopedResourceClient<DashboardDataDTO>({
+      group: 'dashboard.grafana.app',
+      version: 'v0alpha1',
+      resource: 'dashboards',
+    });
+  }
+
+  saveDashboard(options: SaveDashboardCommand): Promise<unknown> {
+    return this.legacy.saveDashboard(options);
+  }
+
+  async dashboardExists(uid: string): Promise<boolean> {
+    try {
+      const r = await this.client.get(uid);
+      if (r.metadata.name) {
+        return Promise.resolve(true);
+      }
+    } catch {}
+    return Promise.resolve(false);
+  }
+
+  deleteDashboard(uid: string, showSuccessAlert: boolean): Promise<DeleteDashboardResponse> {
+    return this.legacy.deleteDashboard(uid, showSuccessAlert);
+  }
+
+  async getDashboardDTO(uid: string): Promise<DashboardDTO> {
+    const d = await this.client.get(uid);
+    const m = await this.client.subresource<object>(uid, 'access');
```

Comment:
I wonder about this line, it's resolving metadata, but what is `access` ? :)

Reply:
The existing api mixes standard "resource" calls with properties that change depending on how asks.  It has the dashboard body and authz as a single request.  This won't play nice with standard k8s where a resource needs to be the same regardless who asks.  I am exploring getting the additional access control from an additional call.

![image](https://github.com/grafana/grafana/assets/705951/2cca7a99-4ad3-403e-b1b1-de918b4f90ed)

This call does not yet get all the info moved to the same places, but is pretty close.

Reply:
@ryantxu we are trying to investigate where a TTL for the dashboardDTO would be from the k8s server? is this even in production today? 🤔 

basically; we would want to know the TTL of the dashboardDTO object on the frontend for 'access'

https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/api/dashboard_api.ts#L63